### PR TITLE
dexter: add proof base for set_baker entrypoints

### PR DESCRIPTION
**Pull request by sskeirik**
Fixes: #212

### CODE REVIEW

File: tests/proofs/dexter/dexter-spec.md [hunk 1]
```diff
@@ -49,6 +49,46 @@ If any of the conditions are not satisfied, the call fails.
...
+        <senderaddr> Sender </senderaddr>
+        <freezeBaker> false => FreezeBaker </freezeBaker>
+        <nonce> #Nonce(N => N +Int 1) </nonce>
+    requires Amount ==Int 0
+```
+
+If any of the conditions are not satisfied, the call fails.
+
+```k
+  claim <k> #runProof(true, SetBaker(_Baker, _NewFreezeBaker)) => Aborted(?_, ?_, ?_, ?_) </k>
```
Comment (hjorthjort) [inline tests/proofs/dexter/dexter-spec.md:78]: _IsFA2 = true here, is that intentional?

Comment (sskeirik) [reply tests/proofs/dexter/dexter-spec.md:78]: No. I'll fix that.

File: tests/proofs/dexter/dexter-spec.md [hunk 2]
```diff
@@ -49,6 +49,46 @@ If any of the conditions are not satisfied, the call fails.
...
+
+1.  the token pool is _not_ currently updating (i.e. `storage.selfIsUpdatingTokenPool = false`)
+2.  exactly 0 tez was transferred to this contract when it was invoked
+3.  the txn sender is the `storage.manager`
+4.  the baker is _not_ already frozen
+
+FIXME: This claim is incomplete because it depends on the `operations` cell which we do not have yet.
+
+```
+  claim <k> #runProof(_IsFA2, SetBaker(_Baker, FreezeBaker)) => . </k>
```
Comment (hjorthjort) [inline tests/proofs/dexter/dexter-spec.md:64]: LGTM, but could we just specify the stack change for now until we have the `operations` cell? To make sure the proof goes through.

Comment (hjorthjort) [reply tests/proofs/dexter/dexter-spec.md:64]: nevermind, that would get messy with the storage updates.

**Review by hjorthjort — APPROVED** [review]
LGTM

File: tests/proofs/dexter/dexter-spec.md (hunk 1 above)
Comment (hjorthjort) [reply tests/proofs/dexter/dexter-spec.md:78]: Cool, if it works then go ahead and merge.

Comment (sskeirik) [reply tests/proofs/dexter/dexter-spec.md]: It works. I've pushed up the fix and adding automerge.

**Pull request merged**